exercises/practice/all-your-base/.meta/example.asm

```diff
@@ -0,0 +1,101 @@
+section .text
+global rebase
+
+rebase:
+    ; Argument registers
+    ; %rdi - input digits array
+    ; %esi - number of input digits
+    ; %edx - input base
+    ; %rcx - output array
+    ; %r8d - output base
+
+    xor rax, rax
+    xor rbx, rbx
```

Comment:
We should be saving rbx, and restoring the original value before we return.

https://en.wikipedia.org/wiki/X86_calling_conventions
"If the callee wishes to use registers RBX, RSP, RBP, and R12–R15, it must restore their original values before returning control to the caller."

At the start:
`push rbx`

Before return:
`pop rbx`

Another option would be to not use rbx, and use r11 instead.

Reply:
Omgosh, this was what was causing the failure in the MacOS CI build! Thanks for pointing that out @keiravillekode!